File: app/src/main/java/com/jameskbride/eventbusmvpdemo/main/MainActivityPresenter.java
package com.jameskbride.eventbusmvpdemo.main;

import android.support.annotation.StringRes;

import com.jameskbride.eventbusmvpdemo.R;
import com.jameskbride.eventbusmvpdemo.network.BurritosToGoApi;
import com.jameskbride.eventbusmvpdemo.network.Order;
import com.jameskbride.eventbusmvpdemo.network.ProfileResponse;

import java.util.List;

import javax.inject.Inject;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivityPresenter {

    MainActivityView view;
    private BurritosToGoApi burritosToGoApi;

    @Inject
        public MainActivityPresenter(BurritosToGoApi burritosToGoApi) {
        this.burritosToGoApi = burritosToGoApi;
    }

    public void getProfile(String id) {
        Call<ProfileResponse> call = burritosToGoApi.getProfile(id);
        call.enqueue(new  Callback<ProfileResponse>() {
            @Override public void onFailure(Call<ProfileResponse> call, Throwable t) {
                view.displayError(R.string.oops);
            }

            @Override public void onResponse(Call<ProfileResponse> call, Response<ProfileResponse> response) {
                ProfileResponse profileResponse = response.body();
                view.displayProfileDetails(profileResponse);
                List<Order> orderHistory = profileResponse.getOrderHistory();
                displayOrders(orderHistory);
            }
        });
    }

    private void displayOrders(List<Order> orderHistory) {
        if (!orderHistory.isEmpty()) {
            view.displayOrders(orderHistory);
        } else {
            view.displayNoOrders();
        }
    }

    public void setView(MainActivityView view) {
        this.view = view;
    }
}

interface MainActivityView {
    void displayError(@StringRes int message);
    void displayProfileDetails(ProfileResponse profileResponse);
    void displayOrders(List<Order> orderHistory);
    void displayNoOrders();
}

File: app/src/main/java/com/jameskbride/eventbusmvpdemo/main/MainActivityPresenter.kt
package com.jameskbride.eventbusmvpdemo.main

import androidx.annotation.StringRes
import com.jameskbride.eventbusmvpdemo.R
import com.jameskbride.eventbusmvpdemo.network.ProfileApi
import com.jameskbride.eventbusmvpdemo.network.Order
import com.jameskbride.eventbusmvpdemo.network.ProfileResponse
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import javax.inject.Inject

class MainActivityPresenter @Inject constructor(val profileApi: ProfileApi) {
    lateinit var view: MainActivityView

    fun getProfile(id: String) {
        val call: Call<ProfileResponse> = profileApi.getProfile(id)
        call.enqueue(object: Callback<ProfileResponse> {
            override fun onFailure(call: Call<ProfileResponse>?, t: Throwable?) {
                view.displayError(R.string.oops)
            }

            override fun onResponse(call: Call<ProfileResponse>?, response: Response<ProfileResponse>?) {
                val profileResponse = response?.body()!!
                view.displayProfileDetails(profileResponse)
                val orderHistory = profileResponse.orderHistory
                displayOrders(orderHistory)
            }
        })
    }

    private fun displayOrders(orderHistory: List<Order>) {
        if (orderHistory.isNotEmpty()) {
            view.displayOrders(orderHistory)
        } else {
            view.displayNoOrders()
        }
    }
}

interface MainActivityView {
    fun displayError(@StringRes message: Int)
    fun displayProfileDetails(profileResponse: ProfileResponse)
    fun displayOrders(orderHistory: List<Order>)
    fun displayNoOrders()
}